From 2e30e979aa1d4b6ef58c9885537736ed6566e26c Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 20 Feb 2024 18:43:25 +1100
Subject: [PATCH 07/10] move setting of default
 CMAKE_INSTALL_{BIN,INCLUDE,LIB}DIR before first use

also unify spelling of CMAKE_INSTALL_INCLUDEDIR following GNUInstallDirs
defaults, see also CMake docs:
https://cmake.org/cmake/help/latest/command/install.html
---
 CMakeLists.txt     | 24 ++++++++++++------------
 src/CMakeLists.txt |  4 ++--
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 56830cf..03f1589 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,6 +57,18 @@ if((CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
   string(APPEND CMAKE_CXX_FLAGS " -fmacro-prefix-map=${CMAKE_SOURCE_DIR}/=''")
 endif()
 
+if (NOT DEFINED CMAKE_INSTALL_BINDIR)
+  set(CMAKE_INSTALL_BINDIR bin)
+endif()
+
+if (NOT DEFINED CMAKE_INSTALL_LIBDIR)
+  set(CMAKE_INSTALL_LIBDIR lib)
+endif()
+
+if (NOT DEFINED CMAKE_INSTALL_INCLUDEDIR)
+  set(CMAKE_INSTALL_INCLUDEDIR include)
+endif()
+
 if (UNIX)
   include(GNUInstallDirs)
   set(prefix ${CMAKE_INSTALL_PREFIX})
@@ -103,18 +115,6 @@ if (APPLE)
   endif()
 endif()
 
-if (NOT DEFINED CMAKE_INSTALL_BINDIR)
-  set(CMAKE_INSTALL_BINDIR bin)
-endif()
-
-if (NOT DEFINED CMAKE_INSTALL_LIBDIR)
-  set(CMAKE_INSTALL_LIBDIR lib)
-endif()
-
-if (NOT DEFINED CMAKE_INSTALL_INCDIR)
-  set(CMAKE_INSTALL_INCDIR include)
-endif()
-
 # SPDX-License-Identifier: (MIT OR CC0-1.0)
 # Copyright 2020 Jan Tojnar
 # https://github.com/jtojnar/cmake-snips
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index eed204f..a612357 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -331,9 +331,9 @@ install(EXPORT sentencepieceTargets
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/sentencepiece")
 
 install(FILES sentencepiece_trainer.h sentencepiece_processor.h
-  DESTINATION ${CMAKE_INSTALL_INCDIR})
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 if (NOT SPM_PROTOBUF_PROVIDER STREQUAL "internal")
-  install(FILES ${SPM_PROTO_HDRS} DESTINATION ${CMAKE_INSTALL_INCDIR})
+  install(FILES ${SPM_PROTO_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 endif()
 
 file(TO_NATIVE_PATH "${PROJECT_SOURCE_DIR}/data" data_dir)
